### PR TITLE
[12.x] Add Support for Selective Relationship Autoloading in withRelationshipAutoloading

### PR DESCRIPTION
**Pull request by moe-mizrak**
This PR adds support for selective column loading in `withRelationshipAutoloading`, similar to how `with('client:id,name')` works.

Please refer to [withRelationshipAutoloading](https://github.com/laravel/framework/pull/53655)

With this enhancement, you can now specify which columns to load for a relationship, improving query performance by only retrieving the necessary data.

For example:

```php
$orders->withRelationshipAutoloading(['client:id,name']);
```

> [!NOTE]
>
> Backward compatibility is preserved:
> When no columns are specified for a relationship, all columns continue to load by default. Other relationships continue autoloading as expected without needing to be explicitly defined.

### CODE REVIEW

**Comment by moe-mizrak** [comment]
> 📌 **Note:** Looks like working for now, I will check some edge cases and add more advance tests.

File: src/Illuminate/Database/Eloquent/Collection.php [hunk 1]
```diff
@@ -753,11 +769,35 @@ protected function duplicateComparator($strict)
...
+        $callback = function ($tuples) use ($relationMap) {
+            if (! empty($relationMap)) {
+                $relationName = explode(':', $tuples[0][0])[0];
+
+                if (isset($relationMap[$relationName])) {
+                    $tuples[0][0] = $relationMap[$relationName];
+                }
+            }
+
+            $this->loadMissingRelationshipChain($tuples);
```
Comment (moe-mizrak) [inline src/Illuminate/Database/Eloquent/Collection.php:799]: todo: this part will be modified, can be improved overall

**Comment by moe-mizrak** [comment]
Not sure if its worth the complexity it adds up, abondening for now.
